astroid/context.py

```diff
@@ -64,24 +67,20 @@ def __init__(self, path=None, nodes_inferred=None):
         """
         self.callcontext: CallContext | None = None
         """The call arguments and keywords for the given context."""
-        self.boundnode = None
+        self.boundnode: NodeNG | None = None
```

Comment:
Again, are we sure that this is not more than `NodeNG`? I'd prefer to do such changes in a separate PR.